### PR TITLE
resolve [Bug]: Coordinate systems of variables do not get checked against known #3114

### PR DESCRIPTION
**Pull request by RuiheLi**
# Description

Fixes # 3114

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [Yes] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/expression_tree/independent_variable.py [hunk 1]
```diff
@@ -131,6 +131,12 @@ def __init__(
             raise pybamm.DomainError(
                 "domain cannot be particle if name is '{}'".format(name)
             )
+        # check coord_sys against KNOWN_COORD_SYS
+        if coord_sys in KNOWN_COORD_SYS:
+            pass
+        else:
+            raise ValueError(
+                f"Coordinate system is {coord_sys}, not in cartesian, cylindrical polar, spherical polar")
```
Comment (kratman) [inline pybamm/expression_tree/independent_variable.py:139]: I would reverse this:
```suggestion
        if coord_sys not in KNOWN_COORD_SYS:
            raise ValueError(
                f"Coordinate system is {coord_sys}, not in cartesian, cylindrical polar, spherical polar")
```

File: tests/rio/my_test.ipynb
Comment (kratman) [inline tests/rio/my_test.ipynb:1]: Was this added accidentally?

**Comment by agriyakhetarpal** [comment]
Note: this PR is from the `develop` branch of the fork—which could cause problems when fetching/merging further changes from the `develop` branch and is usually not good practice

File: tests/rio/my_test.ipynb
Comment (RuiheLi) [reply tests/rio/my_test.ipynb:1]: Yes, sorry about that.

**Comment by RuiheLi** [comment]
> Note: this PR is from the `develop` branch of the fork—which could cause problems when fetching/merging further changes from the `develop` branch and is usually not good practice

Thanks for the suggestion~

**Comment by RuiheLi** [comment]
I have improved a bit as suggested and created a new branch in my fork called [issue_3114_check_coord](https://github.com/RuiheLi/PyBaMM/tree/issue_3114_check_coord), how can I update the changes here?

**Comment by kratman** [comment]
> I have improved a bit as suggested and created a new branch in my fork called [issue_3114_check_coord](https://github.com/RuiheLi/PyBaMM/tree/issue_3114_check_coord), how can I update the changes here?

At the top where you see:
"wants to merge 3 commits into [pybamm-team:develop](https://github.com/pybamm-team/PyBaMM/tree/develop) from [RuiheLi:develop](https://github.com/RuiheLi/PyBaMM/tree/develop)"

You can edit the branch for the PR there. you just change it from your develop branch to the new one.

**Comment by RuiheLi** [comment]
> > I have improved a bit as suggested and created a new branch in my fork called [issue_3114_check_coord](https://github.com/RuiheLi/PyBaMM/tree/issue_3114_check_coord), how can I update the changes here?
> 
> At the top where you see: "wants to merge 3 commits into [pybamm-team:develop](https://github.com/pybamm-team/PyBaMM/tree/develop) from [RuiheLi:develop](https://github.com/RuiheLi/PyBaMM/tree/develop)"
> 
> You can edit the branch for the PR there. you just change it from your develop branch to the new one.

Hi @kratman  Thank you so much for the kind suggestion. I tried that but it seems that it only allows me to change which branch in pybamm-team I merge to, not which branch from my fork. As seen from the screenshot:
image

**Comment by kratman** [comment]
If your branch is in the same repo then it should work by just clicking on the branch from your side. I know I can do that for my PRs. Did you create a new fork when you made the new branch or just a new branch in the same repo?

**Comment by kratman** [comment]
@RuiheLi can this be closed now that #3394 exists?

**Comment by RuiheLi** [comment]
> @RuiheLi can this be closed now that #3394 exists?

Yes.

**Pull request closed without merging**